File: 0077-combinations/0077-combinations.java
class Solution {
    public List<List<Integer>> combine(int n, int k) {
        List<List<Integer>> result = new ArrayList<>();
        backTracking(1, n, k, new ArrayList<>(), result);
        return result;
    }

    private void backTracking(int start, int n, int k, List<Integer> combination, List<List<Integer>> result) {
        if (combination.size() == k) {
            result.add(new ArrayList<>(combination));
            return;
        }
        for (int i = start; i <= n; i++) {
            combination.add(i);
            backTracking(i + 1, n, k, combination, result);
            combination.remove(combination.size() - 1);
        }
    }
}

File: 0077-combinations/0077-combinations.py
class Solution:
    def combine(self, n: int, k: int) -> List[List[int]]:
        result = []  
        def backTracking(start , combination):
            if len(combination) == k:
                result.append(combination.copy())
                return
            for i in range(start,n + 1):
                combination.append(i)
                backTracking(i + 1, combination)
                combination.pop()
        backTracking(1, [])
        return result
            
        
